onnx/onnx.in.proto

```diff
@@ -417,12 +514,47 @@ message TypeProto {
 
 // #if ONNX-ML
 
-  // repeated T
-  message Sequence {
-  // The type and optional shape of each element of the sequence.
-  // This field MUST be present for this version of the IR.
-    optional TypeProto elem_type = 1;
-  };
+  message Scalar { 
+    // This field MUST NOT have the value of UNDEFINED 
+    // This field MUST be present for this version of the IR. 
+    optional TensorProto.DataType data_type = 1; 
+  }; 
+
+  message Sequence { 
+    // The type and optional shape of each element of the sequence. 
+    // This field MUST be present for this version of the IR. 
+    optional TypeProto elem_type = 1; 
+  }; 
+
+  message Tuple { 
+    // The element types of the tuple. 
+    repeated TypeProto elem_type = 1; 
+  }; 
+
+  message Record { 
+    // The type and shape of each field is described by a ValueInfoProto. 
+    // The field names must be unique. 
+    // This field MUST be present for this version of the IR. 
+    repeated ValueInfoProto fields = 1; 
+  }; 
+
+  message Opaque { 
+    // Domain with empty string means ONNX domain. 
+    optional string domain = 1; 
```

Comment:
Do we need a version number too?

Reply:
Hmmm, maybe. Do you think they will change frequently?

Reply:
May be not. I just thought a consistent approach to handling names, especially qualified names, might be helpful.